src/main/java/duke/parser/Parser.java

```diff
@@ -0,0 +1,117 @@
+package duke.parser;
+
+import java.time.DateTimeException;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.ArrayList;
+
+import duke.DukeException;
+import duke.command.AddCommand;
+import duke.command.Command;
+import duke.command.DeleteCommand;
+import duke.command.ExitCommand;
+import duke.command.FindCommand;
+import duke.command.ListCommand;
+import duke.command.MarkCommand;
+import duke.common.InputChecker;
+import duke.ui.BotUI;
+
+/**
+ * Deals with making sense of the user input.
+ * Contains methods that return Command for execution in Duke class and extracting information from user's raw input.
+ */
+
+public class Parser {
+
+    private static final BotUI UI = new BotUI();
+
+    private static String[] splitInput(String input, String regex) {
+        return input.split(regex, 2);
+    }
+
+    /**
+     * Returns different type of Command according to the user raw input.
+     *
+     * @param rawInput user's raw input.
+     * @return Command to be executed in Duke class.
+     * @throws DukeException - thrown if user command is invalid.
+     */
+    public static Command parse(String rawInput) throws DukeException {
+        try {
+            String[] commandAndDetail = rawInput.split(" ", 2);
+            String command = commandAndDetail[0];
+            if (commandAndDetail.length < 2) {

```

Comment:
Felt that you can improve readability on this by converting this to a guard clause